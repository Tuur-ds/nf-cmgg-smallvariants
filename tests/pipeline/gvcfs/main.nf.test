nextflow_pipeline {

    name "Test Workflow main.nf"
    script "main.nf"

    tag "pipeline"
    tag "pipeline_gvcfs"

    options "-stub"

    test("pipeline_default - gvcfs") {

        when {
            params {
                outdir = "${outputDir}"
                input = "${baseDir}/tests/inputs/samplesheet_gvcf.csv"
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(
                    workflow.stdout,
                    path("${outputDir}")
                        .list()
                        .findAll { !it.toString().endsWith("pipeline_info") }
                        .collect { getRecursiveFileNames(it, "${outputDir}") }
                        .flatten()
                ).match("gvcfs") }
            )
        }

    }

}

def getRecursiveFileNames(fileOrDir, outputDir) {
    if(file(fileOrDir.toString()).isDirectory()) {
        return fileOrDir.list().collect { getRecursiveFileNames(it, outputDir) }
    }
    return fileOrDir.toString().replace("${outputDir}/", "")
}
