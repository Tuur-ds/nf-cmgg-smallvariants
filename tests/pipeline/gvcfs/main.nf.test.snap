{
    "gvcfs": {
        "content": [
            [
                
            ],
            [
                "multiqc/multiqc_report.html",
                "samplesheet.csv",
                "test/Ashkenazim/Ashkenazim.haplotypecaller.bed",
                "test/Ashkenazim/Ashkenazim.haplotypecaller.ped",
                "test/Ashkenazim/Ashkenazim.haplotypecaller.vcf.gz",
                "test/Ashkenazim/Ashkenazim.haplotypecaller.vcf.gz.tbi",
                "test/Ashkenazim/reports/Ashkenazim.haplotypecaller.bcftools_stats.txt",
                "test/Ashkenazim/reports/Ashkenazim.haplotypecaller.html"
            ]
        ],
        "meta": {
            "nf-test": "0.9.1",
            "nextflow": "24.10.0"
        },
        "timestamp": "2024-11-07T15:46:42.423860055"
    }
}